pkg/authorization/authorizer/scope/converter.go

```diff
@@ -244,6 +300,37 @@ func (e clusterRoleEvaluator) ResolveRules(scope, namespace string, clusterPolic
 	return rules, nil
 }
 
+func (e clusterRoleEvaluator) ResolveGettableNamespaces(scope string, clusterPolicyGetter client.ClusterPolicyLister) ([]string, error) {
+	_, scopeNamespace, _, err := e.parseScope(scope)
+	if err != nil {
+		return nil, err
+	}
+	rules, err := e.resolveRules(scope, clusterPolicyGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes := authorizer.DefaultAuthorizationAttributes{
+		APIGroup: kapi.GroupName,
+		Verb:     "get",
+		Resource: "namespaces",
+	}
+
+	errors := []error{}
+	for _, rule := range rules {
+		matches, err := attributes.RuleMatches(rule)
```

Comment:
do we also need to check that one of the following is true:

* `rule.ResourceNames` is empty
* `rule.ResourceNames` contains `scopeNamespace`
* `rule.ResourceNames` contains `"*"`?